mlx/mlx.h

```diff
@@ -8,6 +8,7 @@
 #include "mlx/fft.h"
 #include "mlx/ops.h"
 #include "mlx/random.h"
+#include "mlx/safetensor.h"
```

Comment:
I don't think we want that header exposed by default in the public API. If someone wants those details they can include `mlx/safetensor.h` themselves.